segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/ContainerMetadataUpdateTransaction.java

```diff
@@ -505,11 +501,12 @@ private long generateUniqueSegmentId() {
         // The ContainerMetadata.SequenceNumber is always guaranteed to be unique (it's monotonically strict increasing).
         // It can be safely used as a new unique Segment Id. If any clashes occur, just keep searching up until we find
         // a non-used one.
-        long segmentId = Math.max(this.baseMetadata.getOperationSequenceNumber(), ContainerMetadata.NO_STREAM_SEGMENT_ID + 1);
+        long segmentId = this.realMetadata.getOperationSequenceNumber();
```

Comment:
Another clarification question. I don't fully understand why we use op sequence numbers to be segment ids. Is this use op sequence number just opportunistic or is there any other reason for doing it this way? I guess I don't understand how this logic of assigning ids works.

Following the code flow, I can see that: `preProcessMetadataOperation`-> `assignUniqueSegmentId` -> `generateUniqueSegmentId` in the case the segment has no id in the mapping.